src/Pool.sol

```diff
@@ -141,157 +157,243 @@ abstract contract Pool is Pausable, Ownable, ERC4626 {
-        require(fee != 0, "ADD WITHDRAW FEE: The WithdrawFee fee cannot be 0."); // TO-DO Custom Error Pool_WithdrawFeeZeroError
+        if (blockNumber_ == 0) revert Pool_BlockNumberZeroError();
+        if (fee_ == 0) revert Pool_WithdrawFeeZeroError();
 
         // Add the withdraw fee.
-        _withdrawFees.push(WithdrawFee(blockNumber, fee));
+        _withdrawFees.push(WithdrawFee(blockNumber_, fee_));
     }
 
     /**
      * @dev Updates the withdraw fee.
-     * @param withdrawFeeId The ID of the withdrawFee.
-     * @param blockNumber The block number value.
-     * @param fee The fee value.
+     * @param withdrawFeeId_ The ID of the withdrawFee.
+     * @param blockNumber_ The block number value.
+     * @param fee_ The fee value.
      */
-    function updateWithdrawFee(uint32 withdrawFeeId, uint32 blockNumber, uint32 fee)
+    function updateWithdrawFee(uint32 withdrawFeeId_, uint32 blockNumber_, uint32 fee_)
         external
         onlyOwner
         whenNotPaused
-        withdrawFeeExists(withdrawFeeId)
+        withdrawFeeExists(withdrawFeeId_)
     {
         // Check if the parameters are correct
-        require(blockNumber != 0, "UPDATE WITHDRAW FEE: The WithdrawFee blockNumber cannot be 0."); // TO-DO Custom Error Pool_BlockNumberZeroError
-        require(fee != 0, "UPDATE WITHDRAW FEE: The WithdrawFee fee cannot be 0."); // TO-DO Custom Error Pool_WithdrawFeeZeroError
+        if (blockNumber_ == 0) revert Pool_BlockNumberZeroError();
+        if (fee_ == 0) revert Pool_WithdrawFeeZeroError();
 
         // Update the withdraw fee.
-        _withdrawFees[withdrawFeeId].blockNumber = blockNumber;
-        _withdrawFees[withdrawFeeId].fee = fee;
+        _withdrawFees[withdrawFeeId_].blockNumber = blockNumber_;
+        _withdrawFees[withdrawFeeId_].fee = fee_;
     }
 
     /**
      * @dev Removes the withdraw fee.
-     * @param withdrawFeeId The ID of the withdrawFee.
+     * @param withdrawFeeId_ The ID of the withdrawFee.
      */
-    function removeWithdrawFee(uint32 withdrawFeeId)
+    function removeWithdrawFee(uint32 withdrawFeeId_)
         external
         onlyOwner
         whenNotPaused
-        withdrawFeeExists(withdrawFeeId)
+        withdrawFeeExists(withdrawFeeId_)
     {
         // Remove the withdraw fee.
-        delete _withdrawFees[withdrawFeeId]; // TO-DO Check if leaving a gap could be avoidable or doesnt matter
+        delete _withdrawFees[withdrawFeeId_]; // TO-DO Check if leaving a gap could be avoidable or doesnt matter
     }
 
     /**
      * @dev Adds a new deposit.
-     * @param amount.
+     * @param amount_ Amount to be deposited.
      */
-    function deposit(uint32 amount) external whenNotPaused {
-        /*
-            1. Adds the deposit.
-            2. Charges the deposit fee into the recipient address.
-        */
+    function deposit(uint128 amount_) external whenNotPaused nonReentrant {
```

Comment:
gas: nonReentrant is not needed if you use the checks-effects-interactions pattern.